### DISCUSSION TITLE
Detect markers inside/outside a polygon

### DISCUSSION THREAD

**Discussion by alopezme** [TomSchimansky/CustomTkinter#1289] [General]
Hello,

I am using the example polygon of Switerland becasue I live there :-)
There is a way to detect if another markers is inside or outside a polygon?

**Comment by Wolf-SO** [comment]
It's quite unclear what you mean. 
Can you please rephrase? 
Maybe add a screenshot or image to illustrate the issue you have? 
Thanks.

(Maybe it's better to let someone (or something) translate for you, [DeepL](https://www.deepl.com/translator) is my favorite tool, and I often use it back and forth to get my idea clear.)

**Comment by alopezme** [comment]
Sorry.

I want to be able to test if a marker (CanvasPositionMarker) is inside a polygon (CanvasPolygon).

I will try to do it with this libraries:
https://automating-gis-processes.github.io/CSC18/lessons/L4/point-in-polygon.html

**Comment by alopezme** [comment]
Finally that worked for me:

    from shapely.geometry import Point, Polygon

    def left_click_event(self, coordinates_tuple):
        logging.info("function left_click_event")

        # Check for marker inside polygon
        if Point(coordinates_tuple).within(Polygon(self.switzerland_polygon.position_list)):
            logging.info("Point within Switzerland")
        else:
            logging.info("Point outside Switzerland")
